crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -232,6 +234,8 @@ pub struct CumulocityConverter {
     pub pending_fts_download_operations: HashMap<CmdId, FtsDownloadOperationData>,
 
     pub command_id: IdGenerator,
+
+    partial_entity_store: PartialEntityStore,
```

Comment:
Having both an entity store and a partial entity store might be really confusing.

Wouldn't be better to use a facade to the actual entity store:

```rust
struct CumulocityConverter  {
     entity_store: PartialEntityStore,
}

struct PartialEntityStore {
      entity_store: EntityStore,
      cache: CacheOfPendingRegistration,
}
```

Reply:
Yeah, that would be better. Initially I even had the `PartialEntityStore` inside the `EntityStore` itself. But then I moved it out to reduce the impact on existing APIs. But yeah, I agree, some kind of facade would definitely be better.